stl/inc/iterator

```diff
@@ -119,12 +133,19 @@ class insert_iterator { // wrap inserts into container as output iterator
 public:
     using iterator_category = output_iterator_tag;
     using value_type        = void;
-    using difference_type   = void;
     using pointer           = void;
     using reference         = void;
 
     using container_type = _Container;
 
+#ifdef __cpp_lib_concepts
+    using difference_type = ptrdiff_t;
+
+    insert_iterator() = default;

```

Comment:
out of curiosity: Why isn't this one constexpr (in the standard)

Reply:
I have absolutely no idea. (To be honest, I don't understand why we ever put an explicit `constexpr` on a defaulted default constructor.) These default constructors came from P0896, which crossed with P1032 that added `constexpr` to all of the members of the insert iterators, and apparently the editors decided to put `constexpr` on some subset of the default constructors for reasons not apparent to me.